01-string-manipulation/projects/challenge/bin/challenge.dart

```diff
@@ -89,10 +89,11 @@ void buildingStringsExercise() {
 /// Validate that a credit card number contains only numbers and is
 /// exactly 16 digits long.
 void stringValidationExercise() {
-  final regex = RegExp(r'[0-9]{16}');
+  final regex = RegExp(r'^[0-9]{16}$');
```

Comment:
Have you test it?
It's not working either.

Reply:
Check this:

```
void main() {
  final badRegex = RegExp(r'[0-9]{16}');
  print(badRegex.hasMatch('123r123412341234')); // false [OK]
  print(badRegex.hasMatch('1234123412341234')); // true [OK]
  print(badRegex.hasMatch('123412341234123')); // false [OK]
  print(badRegex.hasMatch('12341234123412345')); // true [NOT OK!!!]

  final goodRegex = RegExp(r'^[0-9]{16}$');
  print(goodRegex.hasMatch('123r123412341234')); // false [OK]
  print(goodRegex.hasMatch('1234123412341234')); // true [OK]
  print(goodRegex.hasMatch('123412341234123')); // false [OK]
  print(goodRegex.hasMatch('12341234123412345')); // false [OK]
}
```